cloud/services/object_storage_objects.go

```diff
@@ -0,0 +1,133 @@
+package services
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	s3manager "github.com/aws/aws-sdk-go-v2/feature/s3/manager"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/s3/types"
+	"github.com/aws/smithy-go"
+
+	"github.com/linode/cluster-api-provider-linode/cloud/scope"
+)
+
+func validateObjectScopeParams(mscope *scope.MachineScope) error {
+	if mscope == nil {
+		return errors.New("nil machine scope")
+	}
+
+	if mscope.S3Client == nil {
+		return errors.New("nil S3 client")
+	}
+
+	if mscope.LinodeCluster.Spec.ObjectStore == nil {
+		return errors.New("nil cluster object store")
+	}
+
+	return nil
+}
+
+func CreateObject(ctx context.Context, mscope *scope.MachineScope, data []byte) (string, error) {
+	if err := validateObjectScopeParams(mscope); err != nil {
+		return "", err
+	}
+	if len(data) == 0 {
+		return "", errors.New("empty data")
+	}
+
+	bucket, err := mscope.GetBucketName(ctx)
+	if err != nil || bucket == "" {
+		return "", errors.New("missing bucket name")
+	}
```

Comment:
We should separate these conditions so we can return the error from GetBucketName if there is one instead of replacing it with this other error